spec/models/internal_post_spec.rb

```diff
@@ -0,0 +1,59 @@
+require "rails_helper"
+
+describe InternalPost do
+  it { should belong_to :user }
+  it { should have_and_belong_to_many :tags }
+  it { should have_many :comments }
+
+  # TODO
+  # it { should have_field(:title) }
+  # it { should have_field(:body_markdown) }
+  # it { should have_field(:body_html) }
+
+  it { should validate_presence_of :user }
+  it { should validate_presence_of :title }
+
+  describe '#notify_twitter' do
+    subject { FactoryGirl.build(:internal_post) }
+
+    it 'posts to Twitter after create' do
+      expect(subject).to receive(:notify_twitter)
+      subject.save!
+    end
+  end
+
+  describe '#username' do
+    subject { FactoryGirl.build(:internal_post) }
+
+    it 'return username of user table' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.